impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java

```diff
@@ -143,14 +152,33 @@ public void addInterceptorBinding(AnnotatedType<? extends Annotation> bindingTyp
 
     @Override
     public <T extends Annotation> AnnotatedTypeConfigurator<T> configureQualifier(Class<T> qualifier) {
-        // TODO WELD-2264
-        return null;
+        checkWithinObserverNotification();
+        AnnotatedTypeConfiguratorImpl<T> configurator = new AnnotatedTypeConfiguratorImpl<>(getBeanManager().createAnnotatedType(qualifier));
+        additionalQualifiers.add(configurator);
```

Comment:
Pls log the extension actions - see for example `BeforeBeanDiscoveryImpl.addQualifier(AnnotatedType<? extends Annotation>)`:

     BootstrapLogger.LOG.addQualifierCalled(getReceiver(), qualifier);